### PR TITLE
FetchContext to hold configs in acquire/release calls

### PR DESCRIPTION
**Pull request by npawar**
A cleanup followup to https://github.com/apache/pinot/pull/7295
Creating a wrapper `FetchContext` to keep params required for prefetch/acquire/release.
Also moving prefetch to InstanceOperator, so that it can be released.

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java [hunk 1]
```diff
@@ -18,26 +18,41 @@
...
 public class InstanceResponseOperator extends BaseOperator<InstanceResponseBlock> {
   private static final String OPERATOR_NAME = "InstanceResponseOperator";
 
   private final Operator _operator;
+  private final List<IndexSegment> _indexSegments;
+  private final List<FetchContext> _fetchContexts;
 
-  public InstanceResponseOperator(Operator combinedOperator) {
+  public InstanceResponseOperator(Operator combinedOperator, List<IndexSegment> indexSegments,
+      List<FetchContext> fetchContexts) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java:39]: (Optional) Use `null` to represent no prefetch?

File: pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java [hunk 2]
```diff
@@ -93,4 +108,16 @@ public static long calTotalThreadCpuTimeNs(long totalWallClockTimeNs, long multi
   public String getOperatorName() {
     return OPERATOR_NAME;
   }
+
+  private void prefetchAll() {
+    for (int i = 0; i < _fetchContexts.size(); i++) {
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java:113]: (nit) put `_fetchContexts.size()` in a local variable

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

File: pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java (hunk 2 above)
Comment (npawar) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/InstanceResponseOperator.java]: fixed

**Pull request merged**